_rules/invalid-form-field-value-36b590.md

```diff
@@ -0,0 +1,192 @@
+- [G84: Providing a text description when the user provides information that is not in the list of allowed values](https://www.w3.org/WAI/WCAG21/Techniques/general/G84)
+- [G85: Providing a text description when user input falls outside the required format or values](https://www.w3.org/WAI/WCAG21/Techniques/general/G85)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `input` element has a [form field error indicator][] that identifies it (by referencing its label) and describes the cause of the error.
+
+```html
+<form>
+	<label for="age">Age (years)</label>
+	<input type="number" id="age" value="0"/>
+	<span id="error">Age must be at least 1</span><br />
+	<input type="button" value="Submit" />
+</form>
+```
+
+#### Passed Example 2
+
+The multiple `input` elements share a [form field error indicator][] that identifies the elements unfilled (by referencing their labels) and describes the cause of the error.
+
+```html
+<form>
+	<h2 id="error">You must fill the name field. You must pick a color.</h2>
+	<label for="name">Name (required)</label>
+	<input type="text" id="name" required />
+	<br />
+	<label for="address">Address</label>
+	<input type="text" id="address" />
+	<p>Pick a color (required)</p>
+	<label><input type="radio" name="color" value="blue" required />Blue</label>
+	<label><input type="radio" name="color" value="yellow" />Yellow</label>
+	<br />
+	<input type="button" value="Submit" />
+</form>
+```
+
+#### Passed Example 3
+
+The `input` element does not have a [form field error indicator][].
+
+```html
+<form>
+	<label for="age">Age (years)</label>
+	<input type="number" id="age" />
+	<input type="button" value="Submit" />
+</form>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The multiple `input` elements share a [form field error indicator][] but its message does not identify the elements that caused the error nor describes the cause of the error.
+
+```html
+<form>
+	<div id="error">Please fill the field correctly.</div>
+	<label for="age">Age (years)</label>
+	<input type="number" id="age" />
+	<label for="name">Name</label>
+	<input type="text" id="name" />
+	<input type="button" value="Submit" />
+</form>
+```
+
+#### Failed Example 2
+
+The `input` elements has a [form field error indicator][] but its message does not describe the cause of the error.
+
+```html
+<form>
+	<label for="age">Age (years)</label>
+	<input type="number" id="age" />
+	<span id="error">Please enter a correct age.</span><br />
+	<input type="button" value="Submit" />
+</form>
+```
+
+#### Failed Example 3
+
+The `input` element has a [form field error indicator][] that identifies it (by referencing its label) and describes the cause of the error but the message is not [visible][].
+
+```html
+<form>
+	<label for="age">Age (years)</label>
+	<input type="number" id="age" value="0" />
+	<span id="error" style="display: none;">Age must be at least 1</span><br />
```

Comment:
This does not describe the cause of the error.

But more importantly, the "display: none;" method of making things invisible also(!) removes it from the accessibility tree, so this would fail on multiple accounts. This example would probably be made more clear by making the text invisible but still available in the accessibility tree instead.

Reply:
There is an `aria-describedby` attribute in the submit button that makes the error message available in the accessbility tree.

Reply:
You are correct! I am not sure how I missed that.

That only leaves the "This does not describe the cause of the error" comment I guess.

Reply:
I had already updated the error message. It should describe the cause of error, but let me know if you don't agree.